### ISSUE TITLE
Ultralytics YOLO11 Launch 🚀

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/ultralytics#16602]
## Ultralytics YOLO11

We are thrilled to announce the official launch of **YOLO11**, the latest iteration of the [Ultralytics](https://ultralytics.com) YOLO series, bringing unparalleled advancements in real-time object detection, segmentation, pose estimation, and classification. Building upon the success of [YOLOv8](https://docs.ultralytics.com/models/yolov8), [YOLO11](https://docs.ultralytics.com/models/yolo11) delivers state-of-the-art performance across the board with significant improvements in both speed and accuracy.

![YOLO11 Performance Plots](https://github.com/user-attachments/assets/a311a4ed-bbf2-43b5-8012-5f183a28a845)

### 🚀 Key Performance Improvements:

- **Accuracy Boost:** YOLO11 achieves up to a **2% higher mAP** (mean Average Precision) on COCO for object detection compared to YOLOv8.
- **Efficiency & Speed:** It boasts **up to 22% fewer parameters** than YOLOv8 models while improving real-time inference speeds by up to **2% faster**, making it perfect for edge applications and resource-constrained environments.
  
### 📊 Quantitative Performance Comparison with YOLOv8:

| Model           | YOLOv8 mAP<sup>val</sup> (%) | YOLO11 mAP<sup>val</sup> (%) | YOLOv8 Params (M) | YOLO11 Params (M) | Improvement |
|-----------------|------------------------------|------------------------------|-------------------|-------------------|-------------|
| **YOLOn**       | 37.3                         | 39.5                         | 3.2               | 2.6               | +2.2% mAP   |
| **YOLOs**       | 44.9                         | 47.0                         | 11.2              | 9.4               | +2.1% mAP   |
| **YOLOm**       | 50.2                         | 51.5                         | 25.9              | 20.1              | +1.3% mAP   |
| **YOLOl**       | 52.9                         | 53.4                         | 43.7              | 25.3              | +0.5% mAP   |
| **YOLOx**       | 53.9                         | 54.7                         | 68.2              | 56.9              | +0.8% mAP   |

Each variant of YOLO11 (n, s, m, l, x) is designed to offer the optimal balance of speed and accuracy, catering to diverse application needs.

### 🚀 Versatile Task Support
YOLO11 builds on the versatility of the YOLO series, handling diverse computer vision tasks seamlessly:

- [**Detection**](https://docs.ultralytics.com/tasks/detect/): Rapidly detect and localize objects within images or video frames.
- [**Instance Segmentation**](https://docs.ultralytics.com/tasks/segment/): Identify and segment objects at a pixel level for more granular insights.
- [**Pose Estimation**](https://docs.ultralytics.com/tasks/pose/): Detect key points for human pose estimation, suitable for fitness, sports analytics, and more.
- [**Oriented Object Detection (OBB)**](https://docs.ultralytics.com/tasks/obb/): Detect objects with an orientation angle, perfect for aerial imagery and robotics.
- [**Classification**](https://docs.ultralytics.com/tasks/classify/): Classify whole images into categories, useful for tasks like product categorization.

### 📦 Quick Start Example

To get started with YOLO11, install the latest version of the Ultralytics package:

```bash
pip install ultralytics>=8.3.0
```

Then, load the pre-trained YOLO11 model and run inference on an image:

```python
from ultralytics import YOLO

# Load the YOLO11 model
model = YOLO("yolo11n.pt")

# Run inference on an image
results = model("path/to/image.jpg")

# Display results
results[0].show()
```

With just a few lines of code, you can harness the power of YOLO11 for real-time object detection and other computer vision tasks.

### 🌐 Seamless Integration & Deployment

YOLO11 is designed for easy integration into existing workflows and is optimized for deployment across a variety of environments, from edge devices to cloud platforms, offering unmatched flexibility for diverse applications. 

You can get started with YOLO11 today through the [Ultralytics HUB](https://hub.ultralytics.com) and the [Ultralytics Python package](https://docs.ultralytics.com/quickstart/). Dive into the future of computer vision and experience how YOLO11 can power your AI projects! 🚀

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for sharing the exciting news about Ultralytics YOLO11 🚀! We are delighted to hear about the advancements and improvements this latest iteration offers. 

For those new to Ultralytics, we recommend checking out our [Docs](https://docs.ultralytics.com) for comprehensive guidance and examples for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. This can help answer many common queries and get you started quickly.

If you encounter any 🐛 bugs while exploring YOLO11, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to aid in debugging.

Interested in custom training or have specific questions ❓? Please provide detailed information, including dataset samples and training logs, and ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Connect with the Ultralytics community in the spot that suits you best. Jump into real-time chat on [Discord](https://ultralytics.com/discord) 🎧. Prefer longer discussions? Visit [Discourse](https://community.ultralytics.com). And for engaging threads, our [Subreddit](https://reddit.com/r/ultralytics) awaits!

## Upgrade

Make sure to upgrade to the latest version of `ultralytics` along with all necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be run in these verified environments with dependencies like [CUDA](https://developer.nvidia.com/cuda), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a> 

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, verifying functionality across different systems daily and with every commit.

Thank you for your contribution and enthusiasm! Please note this is an automated response, and an Ultralytics engineer will assist you soon. 🎉

**Comment by RizwanMunawar** [comment]
Here are some cool visualizations with **Ultralytics YOLO11** 🎉


https://github.com/user-attachments/assets/b45293e9-6d4c-4b40-85ab-3da09eea3cac

https://github.com/user-attachments/assets/b51d162e-9ccf-455a-a484-ac644128fa38

https://github.com/user-attachments/assets/cfad9b4a-4903-489b-b3ba-696fb8715c7c

https://github.com/user-attachments/assets/789e514c-ae25-4864-999d-58e7ef3c2481

https://github.com/user-attachments/assets/81b0a516-8837-44ee-aa76-35887e16b366

https://github.com/user-attachments/assets/ba19dd28-cfc6-4048-a7ee-c10d32770827

https://github.com/user-attachments/assets/9b8fa658-8af7-4d33-a6ae-6875df013304

**Comment by Maronq** [comment]
I would like to apply to False mode  :' providing a visual output of tracked objects' .
Could you tell me the command ?
![image](https://github.com/user-attachments/assets/fcd62e29-f960-4aef-82e2-5292f36a0b2b)

**Comment by glenn-jocher** [comment]
To display visual output of tracked objects, use the `show=True` argument in your command.

**Comment by kdtxttt** [comment]
Error when running YOLOv8-Action-Recognition
ImportError: cannot import name 'MViT_V1_B_Weights' from 'torchvision.models.video' (D:\miniconda3\envs\yolov8\lib\site-packages\torchvision\models\video\__init__.py)

my environment
Conda version: conda 24.7.1
CUDA is available. Version: 10.2
PyTorch version: 1.10.1+cu102
Using GPU version of PyTorch.

**Comment by glenn-jocher** [comment]
The error indicates an issue with the compatibility of your PyTorch version (1.10.1) and the required `torchvision` models. YOLOv8-Action-Recognition requires a more recent version of PyTorch and `torchvision`. Please update both to the latest versions compatible with your CUDA version (e.g., PyTorch 1.12+). For installation guidance, refer to the [official PyTorch installation page](https://pytorch.org/get-started/locally/).

**Comment by luckyboby12** [comment]
Hello ! I recently considered loading a tri-modal data set based on the yolov8 algorithm framework, including visible and infrared data sets, and a text data set. I don 't know how to load data, can you give me some advice

**Comment by glenn-jocher** [comment]
Thank you for your question! YOLOv8 supports multi-modal data processing, but integrating tri-modal data like visible, infrared, and text will require custom preprocessing and dataset preparation. You can preprocess and align your data into a compatible format (e.g., YOLO-style annotations for images and structured embeddings for text) and modify the dataloader accordingly. For further customization, refer to the [YOLOv8 training guide](https://docs.ultralytics.com/modes/train/) for dataset preparation details.

**Comment by glenn-jocher** [comment]
The YOLOv11n-seg weights download automatically when using `YOLO('yolov11n-seg.pt')` with ultralytics>=8.3.0. Ensure you:  
1. Use the exact model name `yolov11n-seg.pt` (note the 'v' after yolo)  
2. Have internet connectivity on first run  
3. Upgrade via `pip install -U ultralytics`  

If issues persist, manually download weights from https://docs.ultralytics.com/models/yolo11/#download.

**Comment by GPlays09** [comment]
I'm trying to train on 4 channel (rgba) image data, where do I specify that in config?

**Comment by glenn-jocher** [comment]
@GPlays09 yOLO models process 3-channel RGB images by default. For 4-channel RGBA inputs, you'll need to modify the model architecture in your YAML configuration file to change input channels from 3 to 4 in the first convolutional layer. This modification requires training from scratch rather than using pretrained weights. For implementation details, see the [YOLO model configuration documentation](https://docs.ultralytics.com/usage/cfg/).

**Comment by Syed-bisma2** [comment]
How to improve the YOLOv11s's map.5, map.5-.90, precision, recall as i'm training it on only one class Apple disease detection with a dataset of 9381 annotated images , i'm getting map.5 in 79%, map50-90 in 42%, precision is 63%, recall is 80%, confusion matrix shows 1.00 background, 0.87 of the disease and 0.13 as false negatives...i have trained it for 150 epochs , batch size 32 etc..? please provide a solution to improve the accuracy for only one class..

**Comment by Syed-bisma2** [comment]
How to add attention mechanism in YOLOv11 for improving the accuracy?

**Comment by Syed-bisma2** [comment]
@glenn-jocher What is possible solution for above mentioned questions?

**Comment by glenn-jocher** [comment]
@Syed-bisma2 Thanks for the details—two quick paths that typically help on single-class datasets, and a note on attention:

- To lift mAP/precision: try a slightly larger model or higher resolution and run automated tuning; also calibrate augmentations and your inference thresholds.
```bash
# train with a larger model and higher imgsz, with mild aug and cosine LR
yolo detect train model=yolo11m.pt data=apple.yaml imgsz=832 epochs=200 batch=32 cos_lr=True mosaic=0.5 mixup=0.1 close_mosaic=20
# at inference, raise the threshold to trade recall for precision
yolo detect predict model=best.pt source=val/images conf=0.35 iou=0.6
```
For systematic optimization, use `model.tune()`:
```python
from ultralytics import YOLO
model = YOLO("yolo11s.pt")
model.tune(data="apple.yaml", epochs=30, iterations=150)  # finds better lr/aug/etc.
```
See the Hyperparameter Tuning Guide and Augmentation Guide for what to try next. ([docs.ultralytics.com](https://docs.ultralytics.com/guides/hyperparameter-tuning/))

- About “adding attention” to YOLO11: not supported out of the box and generally hurts speed; if you still want to experiment, you’ll need a custom model YAML and module/backbone swap, then train from scratch. The Model YAML Configuration Guide shows how to plug in custom or TorchVision backbones. ([docs.ultralytics.com](https://docs.ultralytics.com/%5C/guides/model-yaml-config/))

If you can share a short training log (PR/ROC curves) we can suggest tighter changes.

**Comment by Syed-bisma2** [comment]
@glenn-jocher 

<img width="1135" height="852" alt="Image" src="https://github.com/user-attachments/assets/e17a4b8a-9515-4322-912c-96cf8370cba0" />

**Comment by Syed-bisma2** [comment]
@glenn-jocher 

<img width="963" height="795" alt="Image" src="https://github.com/user-attachments/assets/d4a5cbfe-5b40-4cb4-b7c5-ba55e700b67e" />

**Comment by glenn-jocher** [comment]
@Syed-bisma2 thanks for the plots—next, please validate your `best.pt` with TTA and a low `conf` to inspect the full PR/F1 curves, then pick the operating `conf` that maximizes F1; for a quick check run:
```bash
yolo detect val model=best.pt data=apple.yaml imgsz=832 conf=0.001 iou=0.7 augment=True plots=True
```
At inference, test the calibrated thresholds (and TTA) like:
```bash
yolo detect predict model=best.pt source=val/images conf=0.45 iou=0.6 augment=True
```
If small-object FNs dominate, retrain with multi-scale enabled and slightly lighter aug; also share text logs so we can be precise: the last 20 lines of `runs/detect/train*/results.csv`, your `runs/detect/train*/args.yaml`, and `yolo checks` output; references: [Val mode](https://docs.ultralytics.com/modes/val/) and the [Prediction guide](https://docs.ultralytics.com/modes/predict/).

**Comment by Syed-bisma2** [comment]
@glenn-jocher 

[args.yaml](https://github.com/user-attachments/files/23451418/args.yaml)

[csv.xlsx](https://github.com/user-attachments/files/23451964/csv.xlsx)

**Comment by glenn-jocher** [comment]
@Syed-bisma2 thanks for sharing the files—could you paste the last 20 lines of `results.csv` and the text of your `args.yaml` inline, plus the output of `yolo checks`? For single-class cases, try a quick rerun with single-class and multi-scale training and a lighter class loss, then compare PR/F1:

```bash
yolo detect train model=yolo11m.pt data=apple.yaml imgsz=896 epochs=200 batch=32 single_cls=True multi_scale=True cos_lr=True mosaic=0.3 mixup=0.0 close_mosaic=10 cls=0.2 patience=50
```

If you recently upgraded packages and are in a notebook, please restart the kernel before rerunning.

**Comment by glenn-jocher** [comment]
@Syed-bisma2 thanks—your args show `single_cls=False`, `imgsz=640`, `mosaic=1.0`, and `auto_augment=randaugment/erasing=0.4`, which often depress precision on single‑class box tasks; try this targeted rerun and then share `yolo checks` (or `!yolo checks` in notebooks) plus the plain‑text tail of `results.csv` for review:
```bash
yolo detect train model=yolo11m.pt data=E:\Datasets\root_directory\data.yaml imgsz=832 epochs=200 batch=32 \
single_cls=True multi_scale=True rect=True cos_lr=True mosaic=0.4 auto_augment=None erasing=0.0 cls=0.2 close_mosaic=15
```
Also confirm your `data.yaml` has `nc: 1` and, if you recently edited labels, delete any `*.cache` files under the `labels` directories; for automated hyperparameter search, see the [Tune mode guide](https://docs.ultralytics.com/modes/).

**Comment by Syed-bisma2** [comment]
@glenn-jocher Sir, with imgsz=832 and batch_size=32, it shows a CUDA error...CUDA out of memory, as I'm having only two GPUs. What is the solution?

**Comment by glenn-jocher** [comment]
@Syed-bisma2 the CUDA out-of-memory just means that `yolo11m` at `imgsz=832` and `batch=32` is too heavy for your available VRAM; on two GPUs, either reduce the batch (try 16 → 8 → 4) at 832, or keep `batch=32` and lower `imgsz` (e.g., 704 or 640), and optionally set `device=0,1` to use both GPUs—if it still fails after dialing those down, please paste the full CUDA error text and your `yolo checks` output so we can see your exact setup.

**Comment by ghzlnasirijozani-glitch** [comment]
Hey there, I'm New to DL and my project is to add CBAM to layer 3 and 6 of backbone of model Yolov11m and I don't know how to do that can someone help please.

**Comment by ghzlnasirijozani-glitch** [comment]
my project is to modify Yolo v11m for detection, classification and counting blood cells and I taught adding CBAM might be a good improvement but I'm not sure where should I add it to get the best result , would layer 3 and 6 be good?

**Comment by glenn-jocher** [comment]
@ghzlnasirijozani-glitch nice project idea. CBAM isn’t built into YOLO11, so there’s no config flag like `attention=cbam`; you’d need to fork the repo, create a custom YOLO11m YAML, implement a `CBAM` PyTorch module, wire it into the backbone, and then train from scratch using that YAML (no pretrained weights).

A typical workflow is: copy the existing YOLO11m model YAML to something like `yolo11m_cbam.yaml`, add CBAM blocks after the backbone entries you want (e.g., after the 3rd and 6th backbone layers in that file), and then train with:

```bash
yolo detect train model=yolo11m_cbam.yaml data=your_data.yaml epochs=... imgsz=... 
```

There isn’t a guaranteed “best” place for CBAM; you usually experiment with a small number of mid‑depth backbone locations where feature maps are smaller but still contain useful detail, and then compare validation mAP and speed to the vanilla YOLO11m baseline. Also be aware that attention blocks increase latency and memory, and in our experience attention‑heavy variants (like the community YOLO12 models described in the [YOLO12 overview](https://docs.ultralytics.com/models/yolo12/)) can be less stable and slower, so for production we generally recommend sticking with plain YOLO11 and focusing first on data quality, augmentation, and hyperparameter tuning. For background on the supported architectures and how they’re structured, the [Ultralytics model docs](https://docs.ultralytics.com/models/) are a good reference.

**Comment by ghzlnasirijozani-glitch** [comment]
Yes I understand that attention blocks increase latency but i gotta come up with some kind of change for my project and also in this field that I'm working time is not as important as accuracy.
I also have another question.
I want to add some kinda algorithm for rechecking the confidence scores and then sending the ones with low scores to feature map from layer 6 for double checking.
Where should I add this ?
When i want to go for the test for my model should I add this there?

**Comment by glenn-jocher** [comment]
@ghzlnasirijozani-glitch good idea conceptually—what you’re describing is basically a small second-stage “recheck” on low‑confidence boxes, and the cleanest way is to implement it in your own inference script rather than inside the model definition. For example:

```python
from ultralytics import YOLO

model = YOLO("yolo11m.pt")

results = model("image_or_video_frame.jpg")
r = results[0]

# pick boxes below some confidence for extra processing
low_conf_mask = r.boxes.conf < 0.4
low_conf_boxes = r.boxes[low_conf_mask]

for b in low_conf_boxes:
    xyxy = b.xyxy  # box coords
    conf = float(b.conf)
    cls  = int(b.cls)
    # TODO: your custom “double‑check” logic here
```

You can run this during testing/validation or any custom “eval” script; if you truly need layer‑6 feature maps you’d have to fork the repo, modify the YOLO11m `forward()` to return that intermediate feature along with detections, then use it in your custom post‑processing, which is an advanced change and not exposed via config. For more end‑to‑end Python usage patterns (where this logic would live), see the [Ultralytics Python API examples](https://docs.ultralytics.com/usage/python/).

**Comment by ghzlnasirijozani-glitch** [comment]
if I already have my custom YOLO11m YAML with CABM add to it, can I use that for my pro-processing double check step to go back to feature map made by CABM in layer 6 to check through IOU?

**Comment by glenn-jocher** [comment]
Yes, you can use your custom YOLO11m+CBAM model for that, but the YAML alone isn’t enough—you need to expose the layer‑6 CBAM feature map during inference (or evaluation) and run your IoU “double‑check” there. A common way is to attach a forward hook to that layer in a custom Python script:

```python
from ultralytics import YOLO

model = YOLO("yolo11m_cbam.pt")  # your trained CBAM model

feat = {}

def save_layer6_feat(_, __, output):
    feat["layer6"] = output

# adjust the index/path below to match where your CBAM(6) lives in the model graph
layer6 = model.model.model[6]
layer6.register_forward_hook(save_layer6_feat)

results = model("image.jpg")  # normal inference
layer6_map = feat["layer6"]   # CBAM layer-6 feature map for your custom IoU logic
```

This kind of hook‑based “second stage” is best kept in your own eval/inference script (not inside the training/CLI pipeline), so you can freely experiment without fighting the built‑in `val`/`predict` implementations.

**Comment by ghzlnasirijozani-glitch** [comment]
that actually helped a lot, thank you for your time.

**Comment by glenn-jocher** [comment]
Glad it was useful—if you hit any issues while wiring CBAM or the second‑stage check into YOLO11m, share a small reproducible snippet or error trace here and we can help debug.

**Comment by glenn-jocher** [comment]
@Syed-bisma2 in Ultralytics detection the confusion matrix always includes an extra “background” row/column to account for **unmatched boxes** (it’s not a real class you annotate, and the model isn’t “predicting background” as an object class); false positives land in the **background row** (predicted disease with no matching GT), and false negatives land in the **background column** for your class (missed GT disease), so with a single class it’s normal for the plot to look dominated by background—see the notes around metrics/plots in [Model evaluation insights](https://docs.ultralytics.com/guides/model-evaluation-insights/); if you paste the text output of `yolo detect val ...` plus `yolo checks` (and confirm `nc: 1` in `data.yaml`, and delete any `labels/*.cache` after label edits), I can tell you exactly whether that 1.00 is simply normalization or indicates a threshold/label issue.

**Comment by glenn-jocher** [comment]
@Syed-bisma2 In your thesis you can state that for object detection the confusion matrix is extended with a bookkeeping “background” row/column to count unmatched predictions (false positives) and unmatched ground-truth objects (false negatives), so “background=100%” is typically a normalization/imbalance artifact on single-class datasets rather than a learned class; for reporting, emphasize COCO-style metrics (mAP/PR/F1) as primary and include the confusion matrix as a qualitative error breakdown with your exact matching settings (IoU threshold and confidence threshold) documented—see the [confusion matrix definition](https://www.ultralytics.com/glossary/confusion-matrix) for a citable explanation, and please keep follow-ups here on GitHub (we can’t support via email/DM).

**Comment by Syed-bisma2** [comment]
@glenn-jocher Thank you for your time and help. I had done it on GitHub, email was sent by mistake!

**Comment by glenn-jocher** [comment]
No worries at all—thanks for confirming; if anything about the confusion matrix still seems off, paste the plain-text output from your `yolo detect val ...` run (not a screenshot) and your `yolo checks`, and we’ll sanity-check thresholds, `nc`/class mapping, and label formatting.

**Comment by Syed-bisma2** [comment]
@glenn-jocher (Speed: 0.3ms preprocess, 14.4ms inference, 0.0ms loss, 2.4ms postprocess per image). From this, how to calculate FPS? and what is FP32 precision?

**Comment by glenn-jocher** [comment]
FPS is just the inverse of your **total per-image time**, so add `preprocess + inference + postprocess` (ignore `loss` during inference); with `0.3 + 14.4 + 2.4 = 17.1 ms/image` you’re at roughly **~58 FPS** on the same setup (real video FPS can be lower once decoding/IO/display is included—use [Benchmark mode](https://docs.ultralytics.com/modes/benchmark/) if you want a cleaner apples-to-apples throughput number).  

`FP32` means **32-bit floating point (“full precision”)** compute/weights; it’s the standard precise mode, while `FP16`/`BF16` are lower-precision modes often used to speed up GPU inference/training (sometimes with a small accuracy trade-off).  

```python
ms = 0.3 + 14.4 + 2.4
fps = 1000 / ms
fps
```

**Comment by Syed-bisma2** [comment]
@glenn-jocher Hi there, How to calculate how much data has increased due to augmentation techniques? Is there any specific formula for calculating it?

**Comment by glenn-jocher** [comment]
@Syed-bisma2 if you’re using Ultralytics YOLO training augmentations (mosaic/mixup/flips/color, etc.), they’re applied **on-the-fly** by the dataloader, so your dataset size on disk doesn’t “increase” and there isn’t a single fixed formula for “% data increase” because the transforms are stochastic (effectively unlimited unique views); the only clean quantity you can report is how many **augmented samples were seen during training**, which is roughly `N_images × epochs` (and for mosaic you can mention each training sample may combine ~4 source images, increasing *exposure* but not creating new labeled files) as described in the [Train mode docs](https://docs.ultralytics.com/modes/train/).

**Comment by glenn-jocher** [comment]
@Syed-bisma2 With Ultralytics YOLO the default augmentations are applied online, so most papers don’t report a fixed “dataset increased to X images” unless they generated augmented files offline (in which case they just count the new files); for online augmentation, the clean way to write it is in terms of *effective exposures* (roughly `N_train × epochs`, with a note that `mosaic` can combine multiple source images per sample) and by stating which augmentation settings were used—see the [YOLO data augmentation guide](https://docs.ultralytics.com/guides/yolo-data-augmentation/) for the exact controls. For the plot you shared, the “instances” correspond to the split that was scanned to make that plot (during `train` it’s the training split only, not train+val+test).

**Comment by Syed-bisma2** [comment]
@glenn-jocher Thank you so much for your time and help.

**Comment by glenn-jocher** [comment]
You’re welcome—glad it helped, and credit to the broader YOLO community and Ultralytics team; if any other questions come up while writing results or debugging runs, just drop them here with the exact `yolo ...` command plus `yolo checks` output so we can sanity-check quickly.

**Comment by Syed-bisma2** [comment]
@glenn-jocher Hey! Hope you are doing well. Can you explain that I have used the pt. yolov11s , so i have trained the model through transfer learning, how exactly I have used this learning approach, like what layers or block of YOLOv11s are frozen, yolo shows only this "Freezing layer 'model.23.dfl.conv.weight'.

**Comment by glenn-jocher** [comment]
@Syed-bisma2 Training from `yolo11s.pt` is transfer learning in the usual sense: you initialize from the pretrained checkpoint and then fine-tune (by default) essentially the whole network on your dataset; the “Freezing layer `model.23.dfl.conv.weight`” line is normal because the DFL integral conv is constructed with `requires_grad_(False)` by design, so it’s always kept fixed. ([github.com](https://github.com/ultralytics/ultralytics/issues/16903)) If you want to explicitly freeze additional layers (e.g., backbone), set `freeze=` (int or layer index list) as described in the [Ultralytics configuration reference (`freeze`)](https://docs.ultralytics.com/usage/cfg/). ([docs.ultralytics.com](https://docs.ultralytics.com/usage/cfg/))

```bash
yolo detect train model=yolo11s.pt data=your_data.yaml freeze=10
```

**Comment by Syed-bisma2** [comment]
@glenn-jocher Hi! Hope you're doing well! ....The below fig of the dataset for YOLOv11 shows 30, 000 instances of the class Alternaria. Does this count include augmented data points, and is this of the training set only, or does it encompass all three splits?
<img width="569" height="531" alt="Image" src="https://github.com/user-attachments/assets/796271e3-f5b7-4424-be35-7d4c417beb98" />

**Comment by glenn-jocher** [comment]
@Syed-bisma2 That `30,000` is the raw annotation count for the split being scanned, so in a `train` run it’s the training set only; on-the-fly augmentation doesn’t add extra stored instances unless you exported augmented files offline.
